File: python/Scanner.py
from socket import *

def scan(host: str): 
    def checkPort(port):
        try:
            s = socket(AF_INET, SOCK_STREAM)
            s.settimeout(3)
            if s.connect_ex((host, port)) == 0:
                s.close
                return True
        except Exception:
            return False
    
    for p in range(1, 65535):
        if checkPort(p):
            print('port {} is open!'.format(p))

def get_host():
    return input('Enter your host: ')


if __name__ == '__main__':
    host = get_host()
    print('Trying to scan {}'.format(host))
    scan(host)

File: php/Scanner.php
<?php
error_reporting(~E_ALL);

function scanPort($host)
{

    function checkPort($host , $port)
    {
        try {

            $address = "tcp://".trim($host).":$port";
            $timeout = 3;

            $connect = stream_socket_client($address, $errno, $errstr, $timeout);

            if ($connect) {
              fclose($connect);
              return true;
            }
        } catch (Exception $exception) {
            return false;
        }
    }

    foreach(range(1,65535) as $port){
        if (checkPort($host, $port)) {
            echo "port [$port] : open!\n";
        }
    }

}

function getHost()
{
    echo "Please enter the host you want to scan\n";
    $handle = fopen ("php://stdin","r");
    return fgets($handle);
}

$host = getHost();
echo "Trying to scan $host";
scanPort($host);